plugins/modules/zos_lineinfile.py

```diff
@@ -757,16 +757,39 @@ def main():
         if ins_bef:
             stdout = stdout.replace(ins_bef, quotedString(ins_bef))
         try:
+            # Attempt to parse stdout content directly as JSON
             ret = json.loads(stdout)
-        except Exception:
-            messageDict = dict(msg="dsed return content is NOT in json format", stdout=str(stdout), stderr=str(stderr), rc=rc)
-            if result.get('backup_name'):
-                messageDict['backup_name'] = result['backup_name']
-            module.fail_json(**messageDict)
-
-        result['cmd'] = ret['cmd']
-        result['changed'] = ret['changed']
-        result['found'] = ret['found']
+        except json.JSONDecodeError:
+            try:
+                # If parsing fails, clean up possible wrapping quotes
+                if (stdout.startswith("'") and stdout.endswith("'")) or (stdout.startswith('"') and stdout.endswith('"')):
+                    cleaned_stdout = stdout[1:-1]
+                else:
+                    cleaned_stdout = stdout
+                # Replace escaped backslashes with single backslashes
+                cleaned_stdout = cleaned_stdout.replace('\\\\', '\\')
+                # Further clean up: escape any single backslashes not followed by a valid JSON escape character
+                import re
+                cleaned_stdout = re.sub(r'\\([^"\\/bfnrtu])', r'\\\\\1', cleaned_stdout)
+                # Try parsing the cleaned string as JSON
+                ret = json.loads(cleaned_stdout)
+            except Exception:
+                # If still failing, report failure with useful debug info
+                messageDict = dict(
+                    msg="dsed return content is NOT in json format",
```

Comment:
Can you throw the actual error here? One mistake we've been doing in this module so far is obfuscating the error message. Just catch the exception `except Exception as e` and throw the error message here.

```suggestion
                    msg=f"Failed while parsing command output {str(e)} ",
```

Reply:
resolved